lua/starfall/libs_cl/vgui.lua

```diff
@@ -0,0 +1,2295 @@
+local dtxe_methods, dtxe_meta, dtxewrap, dtxeunwrap = instance.Types.UITextEntry.Methods, instance.Types.UITextEntry, instance.Types.UITextEntry.Wrap, instance.Types.UITextEntry.Unwrap
+local dimg_methods, dimg_meta, dimgwrap, dimgunwrap = instance.Types.UIImage.Methods, instance.Types.UIImage, instance.Types.UIImage.Wrap, instance.Types.UIImage.Unwrap
+local dimgb_methods, dimgb_meta, dimgbwrap, dimgbunwrap = instance.Types.UIImageButton.Methods, instance.Types.UIImageButton, instance.Types.UIImageButton.Wrap, instance.Types.UIImageButton.Unwrap
+local dnms_methods, dnms_meta, dnmswrap, dnmsunwrap = instance.Types.UINumSlider.Methods, instance.Types.UINumSlider, instance.Types.UINumSlider.Wrap, instance.Types.UINumSlider.Unwrap
+local dcom_methods, dcom_meta, dcomwrap, dcomunwrap = instance.Types.UIComboBox.Methods, instance.Types.UIComboBox, instance.Types.UIComboBox.Wrap, instance.Types.UIComboBox.Unwrap
+local dclm_methods, dclm_meta, dclmwrap, dclmunwrap = instance.Types.UIColorMixer.Methods, instance.Types.UIColorMixer, instance.Types.UIColorMixer.Wrap, instance.Types.UIColorMixer.Unwrap
+local dmen_methods, dmen_meta, dmenwrap, dmenunwrap = instance.Types.UIMenu.Methods, instance.Types.UIMenu, instance.Types.UIMenu.Wrap, instance.Types.UIMenu.Unwrap
+local dmeno_methods, dmeno_meta, dmenowrap, dmenounwrap = instance.Types.UIMenuOption.Methods, instance.Types.UIMenuOption, instance.Types.UIMenuOption.Wrap, instance.Types.UIMenuOption.Unwrap
+local col_meta, cwrap, cunwrap = instance.Types.Color, instance.Types.Color.Wrap, instance.Types.Color.Unwrap
+local plyunwrap = instance.Types.Player.Unwrap
+local vgui_library = instance.Libraries.vgui
+
+function pnl_meta:__tostring()
+	return "UIPanelBase"
+end
+
+function dpnl_meta:__tostring()
+	return "UIPanel"
+end
+
+function dfrm_meta:__tostring()
+	return "UIFrame"
+end
+
+function dscrl_meta:__tostring()
+	return "UIScrollPanel"
+end
+
+function dbut_meta:__tostring()
+	return "UIButton"
+end
+
+function dlab_meta:__tostring()
+	return "UILabel"
+end
+
+function aimg_meta:__tostring()
+	return "UIAvatarImage"
+end
+
+function dprg_meta:__tostring()
+	return "UIProgress"
+end
+
+function dtxe_meta:__tostring()
+	return "UITextEntry"
+end
+
+function dimg_meta:__tostring()
+	return "UIImage"
+end
+
+function dimgb_meta:__tostring()
+	return "UIImageButton"
+end
+
+function dchk_meta:__tostring()
+	return "UICheckBox"
+end
+
+function dnms_meta:__tostring()
+	return "UINumSlider"
+end
+
+function dcom_meta:__tostring()
+	return "UIComboBox"
+end
+
+function dclm_meta:__tostring()
+	return "UIColorMixer"
+end
+
+function dfb_meta:__tostring()
+	return "UIFileBrowser"
+end
+
+function dmen_meta:__tostring()
+	return "UIMenu"
+end
+
+function dmeno_meta:__tostring()
+	return "UIMenuOption"
+end
+
+local function unwrap(pnl)
+	local pnc = tostring(pnl)
+	local unwrapFunc = instance.Types[pnc].Unwrap
+	
+	return unwrapFunc(pnl)
+end

```

Comment:
This can potentially cause a type to be unwrapped despite not being a panel type.
e.g.
```lua
myPanel.setColor(myPlayer, Color(100,0,0))
```
I'm not sure if the tostring on types can be modified or not, but that would allow even more manipulation of this.

Reply:
I'm 99% sure I did it like that instead of using the specific unwrappers because I ran into a problem when adding a panel as a child to another one.

I'll just do the Panel unwrapper table idea.

Reply:
Done